e2e/kaivuilmoitus_taydennyspyynto.spec.ts

```diff
@@ -0,0 +1,347 @@
+    await page.getByRole('option', { name: `${perustaja.username}` }).click();
+    await page.getByTestId('applicationData.invoicingCustomer.name').fill(asianhoitaja.username);
+    await page.getByTestId('applicationData.invoicingCustomer.registryKey').fill(asianhoitaja.y_tunnus);
+    await page.getByTestId('applicationData.invoicingCustomer.invoicingOperator').click();
+    await page.getByTestId('applicationData.invoicingCustomer.ovt').click();
+    await page.getByTestId('applicationData.invoicingCustomer.ovt').fill('0000000000000');
+    await page.getByTestId('applicationData.invoicingCustomer.invoicingOperator').fill('1');
+    await page.getByRole('button', { name: 'Seuraava' }).click();
+    await page.getByRole('button', { name: 'Seuraava' }).click();
+    await page.getByRole('button', { name: 'Lähetä hakemus' }).click();
+    await page.getByRole('button', { name: 'Vahvista' }).click();
+    await expect(page.getByText('Hakemus lähetetty')).toBeVisible({timeout:30000});
+    await expect(page.locator("[data-testid=related_hanke]")).toBeVisible({timeout:30000})
+    await page.locator("[data-testid=related_hanke]").click();
+    await expect(page.getByText("Hakemukset", {exact:true})).toBeVisible({timeout:10000})
+    await page.getByText("Hakemukset", {exact:true}).click();
+
+    let johtoselvitys = await page.locator("[data-testid^=applicationViewLinkIdentifier-JS]").textContent();
+    let kaivuilmoitus = await page.locator("[data-testid^=applicationViewLinkIdentifier-KP]").textContent();
+    let hanketunnus = await page.locator("[data-testid^=hanke-tunnus]").textContent();
+
+    // käsittely
+    await page.goto(testiData.allu_url);
+    await expect(page.getByPlaceholder('Username')).toBeEmpty();
+    await page.getByPlaceholder('Username').fill(testiData.allupw);
+    await page.getByRole('button', { name: 'Submit' }).click();
+    await expect(page.getByRole('link', { name: 'HAKEMUKSET' })).toBeVisible();
+    await page.getByRole('link', { name: 'HAKEMUKSET' }).click();
+    await expect(page.getByRole('button', { name: 'HAE' })).toBeVisible();
+    await page.getByRole('button', { name: 'HAE' }).click();
+    // kaivuilmoitus
+    await expect(page.getByRole('link', { name: `${kaivuilmoitus}` })).toBeVisible({timeout:20000})
+    await page.getByRole('link', { name: `${kaivuilmoitus}` }).click();
+    await page.getByRole('button', { name: 'NÄYTÄ UUDET TIEDOT' }).click();
+    await page.getByRole('button', { name: 'KÄSITTELYYN' }).click();
+    await page.getByLabel('Hakemuksen lajit *').getByText('Hakemuksen lajit').click();
+    await page.getByText('Katu- ja vihertyöt').click();
+    await page.locator('.cdk-overlay-container > div:nth-child(3)').click();
+    await expect(page.getByRole('button', { name: 'TALLENNA' })).toBeVisible();
+    await page.getByRole('button', { name: 'TALLENNA' }).click();
+    // Johtoselvitys
```

Comment:
I don't think handling the johtoselvitys is necessary for this test. Testing the täydennys for a johtoselvityshakemus is covered by the `johtoselvitys_tilaus_taydennys` test.